src/dowel/logger.py

```diff
@@ -133,29 +135,37 @@
 """
 import abc
 import contextlib
+import re
 import warnings
 
 from dowel.utils import colorize
 
 
 class LogOutput(abc.ABC):
-    """Abstract class for Logger Outputs."""
+    """Abstract class for Logger Outputs.
 
-    @property
-    def types_accepted(self):
-        """Pass these types to this logger output.
+    :param keys_accepted: Regex for which keys this output should accept.
+    """
 
-        The types in this tuple will be accepted by this output.
+    def __init__(self, keys_accepted=r'^$'):
+        self._keys_accepted = keys_accepted
 
-        :return: A tuple containing all valid input types.
-        """
+    @property
+    def types_accepted(self):
```

Comment:
this should probably be an `abc.abstractmethod` rather than returning anything. that way, if a user tries to write a new LogOutput and forgets to implement this method, they will get an error message. Right now, their LogOutput will just reject all logs silently because the base class says so.